src/content/snapshot/troubleshooting-snapshots.md

```diff
@@ -232,6 +232,28 @@ Yes, check out the Chromatic's [modes feature](/docs/modes) is simplifies the pr
 
 </details>
 
+<details>
+
+  <summary>Why are stories with mocked data not being snapshotted correctly?</summary>
+
+Chromatic snapshots sometimes show the initial or intermediate loading state of the UI, instead of the final state with the mocked data. This can lead to visual tests failing, even though the Storybook renders correctly locally. To debug this issue, follow these steps:
+
+1.  Ensure that MSW (Mock Service Worker) is correctly initialized in your Storybook configuration: [here](https://github.com/mswjs/msw-storybook-addon?tab=readme-ov-file#configure-the-addon).
+
+2.  Confirm that the you're not using outdated versions of `msw`, `msw-storybook-addon`, or any other community add-ons.
+
+3.  Run `npm run build-storybook` and `npx http-server storybook-static -o` locally to check for console output and address any MSW warnings or errors. Even if warnings pass locally, they may not work on the Chromatic side.
+
+4.  Pay close attention to how you define MSW handlers, especially for API requests with query parameters. MSW recommends matching only the path in the handler URL and accessing query parameters inside the resolver function using `req.url.searchParams.get()`.
+
+5.  For Interaction tests on Vite, use the Storybook Test addon; on Webpack, use the Storybook Test Runner. Both provide the Test Hook API to help you wait for mocked data and control snapshot timing. This makes debugging mocks easier and avoids capturing loading states. Learn more about the [Test Addon](https://storybook.js.org/docs/writing-tests/test-addon) and [Storybook Test Runner](https://storybook.js.org/docs/writing-tests/test-runner#test-hook-api).
```

Comment:
I don't understand this suggestions. Why would test addon or Test Runner impact how Chromatic runs the story? Chromatic doesn't use either of those tools. 

> Test Hook API to help you wait for mocked data and control snapshot timing
How? AFAIK there is no API to wait for mocked data. 

The only option I'm aware of is the one you recommend in (7) "Assertion that confirms the mocked data is present"